resources/assets/lib/beatmapset-panel.tsx

```diff
@@ -410,151 +467,176 @@ export default class BeatmapsetPanel extends React.Component<Props> {
 
   private renderInfoArea() {
     return (
-      <div className='beatmapset-panel__info'>
-        <div className='beatmapset-panel__info-row beatmapset-panel__info-row--title'>
-          <a className='beatmapset-panel__main-link u-ellipsis-overflow' href={this.url}>
+      <div className="beatmapset-panel__info">
+        <div className="beatmapset-panel__info-row beatmapset-panel__info-row--title">
+          <a
+            className="beatmapset-panel__main-link u-ellipsis-overflow"
+            href={this.url}
+          >
             {getTitle(this.props.beatmapset)}
           </a>
           {this.props.beatmapset.nsfw && <NsfwBadge />}
         </div>
-        <div className='beatmapset-panel__info-row beatmapset-panel__info-row--artist'>
-          <a className='beatmapset-panel__main-link u-ellipsis-overflow' href={this.url}>
-            {osu.trans('beatmapsets.show.details.by_artist', { artist: getArtist(this.props.beatmapset) })}
+        <div className="beatmapset-panel__info-row beatmapset-panel__info-row--artist">
+          <a
+            className="beatmapset-panel__main-link u-ellipsis-overflow"
+            href={this.url}
+          >
+            {osu.trans("beatmapsets.show.details.by_artist", {
+              artist: getArtist(this.props.beatmapset),
+            })}
           </a>
-          {(this.props.beatmapset.track_id != null) && <FeaturedArtistBadge />}
+          {this.props.beatmapset.track_id != null && <FeaturedArtistBadge />}
         </div>
 
-        <div className='beatmapset-panel__info-row beatmapset-panel__info-row--source'>
-          <div className='u-ellipsis-overflow'>
+        <div className="beatmapset-panel__info-row beatmapset-panel__info-row--source">
+          <div className="u-ellipsis-overflow">
             {this.props.beatmapset.source}
           </div>
         </div>
 
-        <div className='beatmapset-panel__info-row beatmapset-panel__info-row--mapper'>
-          <div className='u-ellipsis-overflow'>
+        <div className="beatmapset-panel__info-row beatmapset-panel__info-row--mapper">
+          <div className="u-ellipsis-overflow">
             <StringWithComponent
-              mappings={{ mapper: <MapperLink beatmapset={this.props.beatmapset} /> }}
-              pattern={osu.trans('beatmapsets.show.details.mapped_by')}
+              mappings={{
+                mapper: <MapperLink beatmapset={this.props.beatmapset} />,
+              }}
+              pattern={osu.trans("beatmapsets.show.details.mapped_by")}
             />
+            {this.props.beatmapset.creator !== this.props.currentUserName &&
```

Comment:
compare the profile page user's ID to the beatmapset creator ID. usernames can change and this isn't displaying what you want

... or, you probably won't have to make this check at all if you send the mapsets to browsers in a different field than the other mapsets (which you should)